scipy/signal/wavelets.py

```diff
@@ -308,14 +309,147 @@ def ricker(points, a):
     return total
 
 
-def cwt(data, wavelet, widths):
+def morlet2(M, s, w=5):
+    """
+    Complex Morlet wavelet, designed to work with `cwt`.
+
+    Returns the complete version of morlet wavelet, normalised
+    according to `s`:
+
+        ``exp(1j*w*x/s) * exp(-0.5*(x/s)**2) * pi**(-0.25) * sqrt(1/s)``
+
+    .. versionadded:: 0.18.2
+
+    Parameters
+    ----------
+    M : int
+        Length of the wavelet.
+    s : float
+        Width parameter of the wavelet.
+    w : float, optional
+        Omega0. Default is 5
+
+    Returns
+    -------
+    morlet : (M,) ndarray
+
+    See Also
+    --------
+    morlet2_freq : Conversion between wavelet width and frequency
+
+    Notes
+    -----
+    This function was designed to work with `cwt`. Because `morlet2`
+    returns an array of complex numbers, the `dtype` argument of `cwt`
+    should be set to `numpy.complex` for best results.
+
+    Note the difference in implementation with `morlet`. The fundamental
+    frequency of this wavelet in Hz is calculated using `morlet2_freq`.
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+
+    >>> M = 100
+    >>> s = 4.0
+    >>> w = 2.0
+    >>> wavelet = signal.morlet2(M, s, w)
+    >>> plt.plot(abs(wavelet))
+    >>> plt.show()
+
+    This example shows basic use of `morlet2` with `cwt` in time-frequency
+    analysis:
+
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+    >>> t, dt = np.linspace(0, 1, 200, retstep=True)
+    >>> fs = 1/dt
+    >>> sig = np.cos(2*np.pi*(50 + 10*t)*t) + np.sin(40*np.pi*t)
+    >>> freq = np.linspace(1, fs/2, 100)
+    >>> widths = signal.morlet2_freq(freq, fs, w=6)
+    >>> cwtm = signal.cwt(sig, signal.morlet2, widths, dtype=np.complex, w=6)
+    >>> plt.pcolormesh(t, freq, np.abs(cwtm), cmap='viridis')
+    >>> plt.show()
+
+    """
+    x = np.arange(0, M) - (M - 1.0) / 2
+    x = x / s
+    wavelet = np.exp(1j * w * x) * np.exp(-0.5 * x**2) * np.pi**(-0.25)
+    output = np.sqrt(1/s) * wavelet
+    return output
+
+
+def morlet2_freq(wf, fs, w=5):
```

Comment:
Could be handled by some documentation on this conversion in the Notes section of the ``morlet2`` docstring